Invalid CRON field value 1/2 at position 2

4meck opened the pull request:
dragonmantank\cron-expression (v3.1.0)

$expr = '0 0 1/2 * *';
$cron = \Cron\CronExpression::factory($expr);
echo $cron->getNextRunDate()->format('Y-m-d H:i:s');

InvalidArgumentException in C:\vendor\dragonmantank\cron-expression\src\Cron\CronExpression.phpat line 158
Invalid CRON field value 1/2 at position 2

mtdowling/cron-expression (v. 1.2.3) execute this code correct (w/o Exception)

dragonmantank committed:
Updated README and author information for 2.x

carusogabriel committed:
Update to PHPUnit 6

dragonmantank committed:
Merge pull request #2 from gabriel-caruso/phpunit-6

Update to PHPUnit 6

laurencei committed:
tweak

laurencei committed:
fix capital

laurencei committed:
tweak

dragonmantank committed:
Fixed bad example in README (#7)

dragonmantank committed:
Allow ranges and lists in the same expression (#5)

dragonmantank committed:
Fixed regression where literals were not converted to their numerical values

dragonmantank committed:
Merge pull request #3 from laurencei/timezones

Refactor timezones

dragonmantank committed:
Updated README for 2.1.0

dragonmantank committed:
Resolves #12, added support for numerals with leading 0s

dragonmantank committed:
Resolves #6, added support for ranges to wrap around with large steps

dragonmantank committed:
Updated changelog for 2.2

EntranceJew committed:
fix symmetry in constructor and factory method

if it can be null in the factory, the constructor should have it too

also added phpdoc stuff

dragonmantank committed:
Merge pull request #17 from EntranceJew/patch-1

fix symmetry in constructor and factory method

atefBB committed:
highlight variables

atefBB committed:
highlight variables

phansys committed:
Improve dependency declaration

phansys committed:
Improve dependency declaration

piotr-cz committed:
Get timezone from DateTimeInterface

piotr-cz committed:
Get timezone from DateTimeInterface

piotr-cz committed:
Adding missing docblocks

piotr-cz committed:
Adding missing docblocks

piotr-cz committed:
Adding missing docblocks for class properties

piotr-cz committed:
Adding missing docblocks for class properties

piotr-cz committed:
Normalize EOLs

piotr-cz committed:
Normalize EOLs

EimantasMorkunas committed:
Fixed infinite loop when resolving last weekday of the month from literals

EimantasMorkunas committed:
Fixed infinite loop when resolving last weekday of the month from literals

dragonmantank committed:
Merge pull request #28 from EimantasMorkunas/master

Fixed infinite loop when resolving last weekday of the month from literals

legionth committed:
PHP 7.2 improvement + syntax cleanup + php_cs config update.

dragonmantank committed:
Started listing projects that use cron-expression

dragonmantank committed:
Started listing projects that use cron-expression

dragonmantank committed:
Merge pull request #25 from phansys/dependencies

[dependencies] Improve dependency declaration

dragonmantank committed:
Merge pull request #26 from piotr-cz/bugfix/get-timezone-from-datetimeinterface

Get timezone from DateTimeInterface

dragonmantank committed:
Merge pull request #27 from piotr-cz/cs/docblocks

Add missing docblocks and normalize end of lines

dragonmantank committed:
Positions used higher than 4 now throw a human readable error

dragonmantank committed:
Positions used higher than 4 now throw a human readable error

dragonmantank committed:
Merge pull request #22 from atefBB/patch-1

highlight variables in `README`

dragonmantank committed:
Added PHP 7.3 support, fixed nightly

dragonmantank committed:
Added PHP 7.3 support, fixed nightly

Webonaute committed:
remove php <= 7.1

holtkamp committed:
Use null coalescing operator

holtkamp committed:
Introduce FieldFactoryInterface

holtkamp committed:
Introduce PHPStan

holtkamp committed:
Introduce PHPStan

holtkamp committed:
Reduce strictness level to add tests to analysis

holtkamp committed:
Reduce strictness level to add tests to analysis

cmorbitzer committed:
Support DateTimeImmutable

cmorbitzer committed:
Support DateTimeImmutable

dragonmantank committed:
Merge pull request #41 from cmorbitzer/support-datetimeimmutable

Support DateTimeImmutable

dragonmantank committed:
Merge pull request #39 from holtkamp/patch-adopt-PHPStan

Introduce PHPStan

dragonmantank committed:
Merge branch 'master' into master

dragonmantank committed:
Upped PHP to 7.2

dragonmantank committed:
Merge branch 'master' into master

dragonmantank committed:
Cleaned up some issues from rebasing

dragonmantank committed:
Removed test that checked for invalid character

holtkamp committed:
Move PHPStan to require-dev

piotr-cz committed:
Invalid numeric ranges

GrahamCampbell committed:
Switch to short array syntax

dragonmantank committed:
Merge pull request #43 from holtkamp/patch-1

Move PHPStan to require-dev

dragonmantank committed:
Merge pull request #47 from GrahamCampbell/patch-1

Switch to short array syntax

dragonmantank committed:
Apply fixes from StyleCI

dragonmantank committed:
Merge pull request #48 from dragonmantank/analysis-XZNv26

Apply fixes from StyleCI

dragonmantank committed:
Merge pull request #45 from piotr-cz/bugfix/invalid-numeric-ranges

Handle invalid numeric ranges

dragonmantank committed:
Dropped support back to 7.1 from 7.2

dragonmantank committed:
Added macros we support

dragonmantank committed:
Cleaned up some README formatting, and typo on PHP version

dragonmantank committed:
Verified that 0/5 should now fail

dragonmantank committed:
Fixed #24, incoming literals are automatically converted to uppercase

fabpot committed:
fixed constructor optional argument

fabpot committed:
fixed issue with timezones

fabpot committed:
fixed cloning dates in isDue()

dragonmantank committed:
Merge pull request #49 from fabpot/constructor-fix

Fix constructor optional argument

dragonmantank committed:
Merge pull request #50 from fabpot/timezone-issue-fix

Fix issue with timezones

dragonmantank committed:
Merge pull request #51 from fabpot/clone-fix

Fix cloning dates in isDue()

dragonmantank committed:
Initial fix for and/or problem

dragonmantank committed:
Make sure that shortcuts are not case sensitive

dragonmantank committed:
Removed branch alias, as actual branches should be used not aliases

dragonmantank committed:
Updated changelog for 2.4.0

dragonmantank committed:
Created a FUNDING.yml

GrahamCampbell committed:
Update .travis.yml

dragonmantank committed:
Merge pull request #65 from GrahamCampbell/patch-1

Test on PHP 7.4

PabloKowalczyk committed:
Add Crunz to "Projects that Use cron-expression" section

alexander-schranz committed:
Mark this package as replace for mtdowling/cron-expression

dragonmantank committed:
Merge pull request #68 from PabloKowalczyk/add-crunz

Add Crunz to "Projects that Use cron-expression" section

rgson committed:
Replace deprecated expectedException annotations

From PHPUnit 8:
> The @expectedException, @expectedExceptionCode,
> @expectedExceptionMessage, and @expectedExceptionMessageRegExp
> annotations are deprecated. They will be removed in PHPUnit 9.
> Refactor your test to use expectException(), expectExceptionCode(),
> expectExceptionMessage(), or expectExceptionMessageRegExp() instead.

dragonmantank committed:
Merge pull request #70 from rgson/phpunit-deprecations

Replace deprecated expectedException annotations

dragonmantank committed:
Merge pull request #69 from alexander-schranz/patch-1

Mark this package as replace for mtdowling/cron-expression

holtkamp committed:
Merge branch 'master' of https://github.com/dragonmantank/cron-expression into patch-use-null-coalescing-operator

# Conflicts:
#	src/Cron/FieldFactory.php

holtkamp committed:
Merge branch 'master' of https://github.com/dragonmantank/cron-expression into patch-use-FieldFactoryInterface

holtkamp committed:
Fix CI issues

holtkamp committed:
Sync with master

holtkamp committed:
Actually use FieldFactoryInterface

dragonmantank committed:
Merge pull request #61 from dragonmantank/issue/35-fix-and-or-problem

Initial fix for and/or problem

dragonmantank committed:
Added regression test and cleaned up some ?CHANGELOG language

dragonmantank committed:
Expanded tests for casing

dragonmantank committed:
Merge pull request #38 from holtkamp/patch-use-FieldFactoryInterface

Introduce FieldFactoryInterface

dragonmantank committed:
Merge pull request #37 from holtkamp/patch-use-null-coalescing-operator

Use null coalescing operator

dragonmantank committed:
Added CHANGELOG entry for #38

dragonmantank committed:
StyleCI fixes

dragonmantank committed:
Added missing check for ? in DOM and DOW

dragonmantank committed:
Cast step to an int as part of calling range(), see #80

ausi committed:
Add compatibility with PHP 8.0

ausi committed:
Fix type error

oallain committed:
set phpstan to level max and fix issues

oallain committed:
add phpstan to travis

oallain committed:
fix StyleCI

oallain committed:
fix dependencies for webmozart/assert

dragonmantank committed:
Merge pull request #92 from ausi/fix/php-8-compatibility

Add compatibility with PHP 8.0

dragonmantank committed:
Added support for PHP 8

dragonmantank committed:
Fixed phpstan and phpunit dependencies for v3 line

DerekCresswell committed:
Allow proper usage of interfaces

The constructor for CronExpression should take a FieldFactoryInterface,
not FieldFactory.

dragonmantank committed:
Merge pull request #98 from DerekCresswell/use-interfaces

Allow proper usage of interfaces

DerekCresswell committed:
Depreceate CronExpression::factory and remove references (#99)

* Depreceate CronExpression::factory and remove references
* Update styling

Removed a blank line to make StyleCI happy.

Co-authored-by: Chris Tankersley <chris@ctankersley.com>

DerekCresswell committed:
Adds getParts to CronExpression (#84)

simonschaufi committed:
Ignore files when exporting package (#78)

This commit is part of a campaign to reduce the amount of data transferred to save global bandwidth and reduce the amount of CO2. See https://github.com/Codeception/Codeception/pull/5527 for more info.

DerekCresswell committed:
Updates Phpunit configuration (#102)

Phpunit had output warnings with the outdated configuration. This runs
--migrate-configuration to update it.

oallain committed:
add github action - phpstan and phpunit (#95)

dragonmantank committed:
Merge branch 'master' into phpstan-level-max

dragonmantank committed:
Fixing max run tests for PHP 7.x, excluding 8.0

dragonmantank committed:
Merge branch 'oallain-phpstan-level-max'

dragonmantank committed:
Cleaned up GHA, dropped PHP 7.1 support (#104)

* Allow 8.0 to fail cleanly, because there are some slight differences with DateTime that can't be easily reconciled in phpstan
* Upped minimum version to 7.2
* Redid some of the GHA syntax

holtkamp committed:
Mark YEAR constant as deprecated to prevent usage (#87)

dragonmantank committed:
Added changelog for 3.1.0

DerekCresswell committed:
Change private class variables to protected (#106)

This allows extending classes to access the order for parts the same way
as the base class.

dragonmantank commented:
Hello! Could you please post this PR against dragonmantank/cron-expression, this version of the library is no longer maintained. 

Thanks!

oleksiikhr committed:
refactor(optimize): getMultipleRunDates method (#75)

peter279k committed:
Improve instance of assertion (#105)

BlackIkeEagle committed:
midnight is also an allowed shortcut in crontab (#117)

@see https://manpages.debian.org/buster/cron/crontab.5.en.html

this adds the `@midnight` to the mapping which is the same as `@daily`.
But since it is available in some cron implementations it should not be
indicated as invalid.

Signed-off-by: BlackEagle <ike.devolder@gmail.com>

dragonmantank committed:
F Updated changelog for v3.2.0 release

selfiens committed:
Fix "," delimiter precedence (#122)

dragonmantank committed:
Added changelog for v3.2.1

olafkryus committed:
Fixed getPreviousRunDate return value for cron expression with both day-of-month and day-of-week set (#121)

Commit:
Update tests.yml (#125)

peter279k committed:
Fix failing PHPStan issue (#130)

AllenJB committed:
DST handling fixes (fixes #111, #112) (#115)

* DST fix (attempt 1)

* More tests and further fixes
WIP as I believe the change to HoursField::isSatisfiedBy is going to break expressions with multiple parts - but one thing at a time!

* WIP More tests and further fixes;
API change: isSatisfiedBy now requires knowledge of which direction we're travelling in to handle checking initial values correctly

* WIP Save Point - Trying to fix one thing breaks another, but I have an idea...

* WIP Abstracted NextRunDateTime which keeps track of offset changes, regardless of whether they occurred when changing minute or hour (and to persist them until a next change is made)

* WIP Fix easy fixes

* WIP More test fixes

* All current tests pass!

* Fix for issue #112;
Use a cache of timezone transitions to avoid having to modify date/time objects every single time we want to check if hour is satisfied
All tests pass

* Cleanup
All tests pass

* Cleanup
All tests pass

* Cleanup - backing out the NextRunDateTime abstraction;
All tests pass

* Cleanup;
All tests pass

* Cleanup (diff tidy, restoring deleted tests);
All tests pass

* Cleanup (diff tidy);
All tests pass

* Fix CI issues

* Fix CI issues

Co-authored-by: Chris Tankersley <chris@ctankersley.com>

dragonmantank committed:
Marked a bunch of field stuff @internal

dragonmantank committed:
Fixing #89, wraparound shouldn't check the high value

dragonmantank committed:
Fixes #88, if step is larger than small range only set start

dragonmantank committed:
Added changelog for v3.2.2

dragonmantank committed:
Changed how hours and minutes increment/decrement for better TZ handling

dragonmantank committed:
Added changelog for v3.2.3

dragonmantank committed:
Changed handling of DOM for DST crossover

dragonmantank committed:
Additional tests covering other times that cropped up from v3.2.1 and issue 131

dragonmantank committed:
Fix for DOW DST changes, and updated README

dragonmantank committed:
Fixed run date calculations when ? is used to skip

dragonmantank committed:
Changed requirements from 7.1+ to 7.2+

nyamsprod committed:
Adding Registering alias feature to the package (#132)

dragonmantank committed:
More tests around DOW and DOM resolutions

dragonmantank committed:
Fixed PHPStan errors

dragonmantank committed:
Added changelog for 3.3.0

dragonmantank committed:
Fixed #134, no transitions now immediately returns null

Ayesh committed:
[PHP 8.2] Fix `${var}` string interpolation deprecation (#142)

PHP 8.2 deprecates `"${var}"` string interpolation pattern.
This fixes all three of such occurrences in `dragonmantank/cron-expression` package.

 - [PHP 8.2: `${var}` string interpolation deprecated](https://php.watch/versions/8.2/${var}-string-interpolation-deprecated)
 - [RFC](https://wiki.php.net/rfc/deprecate_dollar_brace_string_interpolation)

athos-ribeiro committed:
Skip PHP 8.1 unstable daylight savings tests (#146)

As discussed in #133, the PHP 8.1's date extension daylight saving
APIs have been suffering with instabilities. Let's skip the affected
tests until https://github.com/php/php-src/issues/9165 is resolved.

Signed-off-by: Athos Ribeiro <athos.ribeiro@canonical.com>

Signed-off-by: Athos Ribeiro <athos.ribeiro@canonical.com>

dragonmantank committed:
Allowed some additional composer plugins to run

dragonmantank committed:
Fixed some PHPStan errors around comparisons and literal positions

dragonmantank committed:
Changelog for 3.3.2

dragonmantank committed:
Added regression tests for issue #151

LeoVie committed:
#137 Add check for multiple question marks (#148)

* #137 Add check for multiple question marks
* #137 Validate position of question marks in expression

imyip committed:
Fix skipped next execution time (#160)

imyip committed:
Fix skipped next execution time (#160)

onemoreangle committed:
Update CronExpression.php (#158)

Added "throws" docblock section, so it's immediately obvious in which way an invalid cron expression will fail

Co-authored-by: Roy Garrido <109597356+r-garrido@users.noreply.github.com>

TimWolla committed:
Mark `phpstan.neon` as `export-ignore` in `.gitattributes` (#144)

erikn69 committed:
Update .gitattributes (#143)

PabloKowalczyk committed:
Change Crunz repository (#139)

dragonmantank committed:
Added Changelog for 3.3.3

sergiy-petrov committed:
Tests against php 8.2 and 8.3 (#167)

xabbuh committed:
explicitly mark nullable parameters as nullable (#175)

Chris53897 committed:
chore: remove travis, and change status badge to github actions (#164)

Co-authored-by: Christopher Georg <christopher.georg@sr-travel.de>

Chris53897 committed:
chore: improve ci, fix deprecations, add php 8.2 tests (#163)

Co-authored-by: Christopher Georg <christopher.georg@sr-travel.de>
Co-authored-by: Chris Tankersley <chris@ctankersley.com>

TimWolla committed:
Remove duplicate `/phpstan.neon` entry in .gitattributes (#162)

This duplicate was not added in alphabetical order and thus probably was missed
during PR merge.

see 6dda2176a2e2092a2169bb7bab87046b8758165a

nicolas-grekas committed:
Add branch-alias to composer.json (#182)

Jubeki committed:
Add PHP 8.4 Support (#190)

8ctopus committed:
Remove phpstan/phpstan-webmozart-assert dependency (#187)

mabar committed:
Docs: full syntax overview (#184)

mabar committed:
Docs: new integrations (#183)

8ctopus committed:
Remove webmozart assert dependency (#192)

Chris53897 committed:
chore: upgrade phpstan, improve docblock for types (#207)

Co-authored-by: Christopher Georg <christopher.georg@sr-travel.de>

Chris53897 committed:
[ci] Add Tests for PHP 8.5 (#205)

* [ci] Add Tests for PHP 8.5

* ignore phpstan errors

---------

Co-authored-by: Christopher Georg <christopher.georg@sr-travel.de>

pierredup committed:
Fix badges in README.md (#198)

CipherdevNL committed:
fix: skipped run after daylight saving turnover point (#203)

Co-authored-by: CipherdevNL <opensource@cipherdev.nl>

Chris53897 committed:
chore: remove PHPStan ignored errors (#208)

Co-authored-by: Christopher Georg <christopher.georg@sr-travel.de>

Chris53897 committed:
chore: bump github actions/checkout 4 => 5 (#210)

Co-authored-by: Christopher Georg <christopher.georg@sr-travel.de>

dragonmantank committed:
Dropped support for PHP older than 8.2 (#211)

Chris53897 committed:
chore: upgrade to PHPUnit 11 and 12 (#212)

Co-authored-by: Christopher Georg <christopher.georg@sr-travel.de>

dereuromark committed:
Add usage in CakePHP plugin (#193)
